src/vsi.rs

```
@@ -7,6 +7,39 @@ use gdal_sys::{VSIFCloseL, VSIFileFromMemBuffer, VSIFree, VSIGetMemFileBuffer, V
 use crate::errors::{GdalError, Result};
 use crate::utils::{_last_null_pointer_err, _path_to_c_string};
 
+/// Read the file names from a virtual file system with optional recursion.
+pub fn read_dir<P: AsRef<Path>>(path: P, recursive: bool) -> Result<Vec<String>> {
+    let path = _path_to_c_string(path.as_ref())?;
+    let mut files = Vec::new();
+    unsafe {
+        // VSIReadDir returns a pointer to C string (null terminated) pointers. The list
+        // of C string pointers is itself also terminated by a null pointer.
+        let data = if recursive {
+            gdal_sys::VSIReadDirRecursive(path.as_ptr())
+        } else {
+            gdal_sys::VSIReadDir(path.as_ptr())
+        };
+
+        let mut index = data as usize;
+        if index != 0 {
+            loop {
+                let ptr = *(index as *mut *mut i8);
+                if ptr as usize == 0 {
+                    break;
+                }
+
+                if let Ok(file) = std::ffi::CStr::from_ptr(ptr).as_ref().to_str() {
+                    files.push(String::from(file));
+                }
```

What is the rationale for ignoring the error vs. returning the error?